### ISSUE TITLE
Storing display settings in Preferences.jl

### ISSUE THREAD

**Issue by bkamins** [JuliaData/DataFrames.jl#3286]
@nalimilan, @ronisbr, @pdeffebach - what would you say if we stored display settings for DataFrames.jl in [Preferences.jl](https://github.com/JuliaPackaging/Preferences.jl)?

Now we use `ENV`, but it is a bit clumsy and not persistent. Additionally then we could allow users to set more preferences.

The downside is that display would be a bit less reproducible.

**Comment by ronisbr** [comment]
It would be awesome! Many years ago I tried to develop a similar system in PrettyTables to store pre-defined keyword parameters. However, when I tried, I could not manage to make it work without affect the TTFX by a huge margin. I think it is time to revisit this :)

**Comment by nalimilan** [comment]
I agree that sounds like a better mechanism than environment variables. Though this part of the docs is a bit worrying:
> Preferences that are accessed during compilation are automatically marked as compile-time preferences, and any change recorded to these preferences will cause the Julia compiler to recompile any cached precompilation .ji files for that module. 

With extensive precompilation, does that mean that changing preferences will require recompilation? That would sound absurd in our case.

**Comment by timholy** [comment]
Indeed, runtime preferences do not force recompilation. Display preferences are much better as runtime preferences (you can change them on-the-fly), so I think you're fine.
